File: 0238-product-of-array-except-self/0238-product-of-array-except-self.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int n= nums.size();
        int curr =1;
        vector<int>ans(n,1);
        for(int i=0;i<n;++i){
            ans[i]=curr;
            curr *=(nums[i]);
        }
        curr=1;
        for(int i=n-1;i>=0;--i){
            ans[i]*=curr;
            curr*=(nums[i]);
        }
        return ans;
    }
};

File: 0238-product-of-array-except-self/0238-product-of-array-except-self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        ans = [0 for i in range(len(nums))]
        cnt = 0
        mul = 1
        for i in nums:
            if i==0:
                cnt+=1
            else :
                mul *=i
        if cnt == 0 :
            for i in range (len(nums)):
                ans[i] = mul//nums[i]
        elif cnt == 1 :
            for i in range(len(nums)):
                if nums[i]== 0:
                    ans[i] = mul
        return ans